### ISSUE TITLE
[Bug]: localhost replaced with [::1]

### ISSUE THREAD

**Issue by debs-obrien** [microsoft/playwright#30858]
### Version

1.44

### Steps to reproduce

run tests then run report

### Expected behavior

should open on localhost:9323

### Actual behavior

Serving HTML report at http://[::1]:9323

### Environment

```shell
System:
    OS: macOS 14.4.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 3.27 GB / 64.00 GB
  Binaries:
    Node: 20.11.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.4 - /usr/local/bin/npm
    pnpm: 8.5.0 - /usr/local/bin/pnpm
  IDEs:
    VSCode: 1.89.1 - /usr/local/bin/code
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/test: ^1.43.0 => 1.44.0
```

**Comment by pavelfeldman** [comment]
Fixed in 1.44.1

**Comment by mxschmitt** [comment]
Fixed in Canary: npm install -D @playwright/test@next
